opensearchpy/helpers/query.py

```diff
@@ -219,10 +220,17 @@ def __and__(self, other: "Bool") -> Any:
                 del q._params["minimum_should_match"]
 
             for qx in (self, other):
-                # TODO: percentages will fail here
                 min_should_match = qx._min_should_match
+
+                # attempt to convert a string integer representation to int
+                with contextlib.suppress(ValueError):
```

Comment:
I am worried that `int()` may do unexpected things like rounding a float for example. 

Maybe we can just be explicit, check that the value ends with % and convert the rest?

Reply:
I think you misunderstood me, sorry. 

I mean that if we rely on an exception to see if the value can be converted, then we might introduce undesired behavior. 

```
$ python -c "print(int("3.14"))"
3
```

You can see that if I pass an incorrect value it's happily coerced into an int, losing the actual value! That doesn't seem good :)